oolab/src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -0,0 +1,67 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+abstract class AbstractWorldMap implements IWorldMap {
+
+    final Vector2d start_vector = new Vector2d(0,0);
+    protected List<Animal> animals = new ArrayList<>();
+    protected List<Grass> grass = new ArrayList<>();
+    protected List<IMapElement> mapElements = new ArrayList<>();
```

Comment:
trzeba by się zdecydować na jedno rozwiązanie - albo mamy osobne kolekcje na różne typy elementów (i wtedy w abstract tylko zwierzęta), albo jedną kolekcję na wszystko. Trzymanie tej samej informacji w dwóch różnych kolekcjach nigdy nie jest dobrym pomysłem, bo prędzej czy później się to rozjedzie.